### ISSUE TITLE
Question: train quotas

### ISSUE THREAD

**Issue by Eldrinn-Elantey** [TwentyEighty/factorio-trainfactory#1]
Hello,

First of all, I want to say that I am glad that someone is dealing with this issue, because I think it would be useful.

My question is whether a quota system for trains will be implemented. Something like a menu where you can specify the number of trains and their type on the railway line, or will it be implemented by a banal limitation of resources for the TrainFactory input?

**Comment by TwentyEighty** [comment]
Here's how I'm currently planning on using it. The train factory keeps building trains as long as they have space on the tracks. The trains are sent to a specified stop. The specified stop is also an LTN depot with a low priority. This means that when LTN runs out of trains to fulfill orders from normal depots, it pulls these new trains. After this happens the tracks clear out and a new train gets built again.

I guess to answer your question, it's limited by a banal lack of room to place the train on the tracks. I'm sure your usecase is much different. If you can describe your idea more and how the system could work I can take a look. What should happen to the train after it's built? What trains should be counted on the railways?

**Comment by Eldrinn-Elantey** [comment]
When I thought about this topic, the first thought was to be able to count the number of empty seats in the LTN depot and for such a plant to build trains until the depot runs out of space. It's just that if there are more trains than the capacity of all depots, there will be traffic jams from trains waiting for a free place in the depot

**Comment by TwentyEighty** [comment]
Yes this was actually one of my issues with TrainConstructionSite. But there's already a solution here
![image](https://user-images.githubusercontent.com/49283746/180800202-b3c7efbd-775b-4e88-9f5e-b2565b0fe21a.png)

Here's one solution (MS Paint warning). This train only gets pulled into the network if you run out of trains in the normal depots. There may be other solutions, but this won't clog the network under normal circumstances. If it does clog the network, it means you need to build more depots anyway to handle the number of jobs your trains are doing.

![image](https://user-images.githubusercontent.com/49283746/180801744-9bde421a-5e27-47ff-b1b5-8a7df8dc5389.png)

**Comment by Eldrinn-Elantey** [comment]
Is it possible to automate the process of installing equipment in the locomotive grid?

**Comment by TwentyEighty** [comment]
It's possible! And planned. It's tricky to make it usable for users though. It's doable to require one building for each equipment like equipment gantry mod does. I'm not sure there's a reasonable way to do it all in one building, because it's not something that can be added to recipes. The equipment could all be added to the same box in one building, but how to actually fill the box with the right set of equipment? For that reason I think it will probably end up being one building for each equipment
